lib/flow_runner/spec/resource.ex

```diff
@@ -24,27 +24,26 @@ defmodule FlowRunner.Spec.Resource do
 
   @spec matching_resource(Resource.t(), language :: String.t(), mode :: String.t(), Flow.t()) ::
           {:ok, ResourceValue.t()} | {:error, String.t()}
-  def matching_resource(%Resource{values: resources}, language, mode, %Flow{languages: languages}) do
+  def matching_resource(%Resource{values: resources}, language, mode, flow) do
     # Identify the language object that corresponds to our iso-639-3 code.
-    languages = Enum.filter(languages, &(&1.iso_639_3 == language))
-
-    if length(languages) > 0 do
-      [language | _] = languages
-      # Filter resources by language and mode.
-      matching_source? = fn x ->
-        x.language_id == language.id && ResourceValue.supports_mode(x, mode)
-      end
-
-      resource_values = Enum.filter(resources, matching_source?)
-
-      if length(resource_values) > 0 do
-        [resource_value | _] = resource_values
-        {:ok, resource_value}
-      else
-        {:error, "no matching resource"}
-      end
+    language =
+      FlowRunner.language_for_context(
+        flow,
+        %FlowRunner.Context{language: language}
+      )
+
+    # Filter resources by language and mode.
+    matching_source? = fn resource ->
+      resource.language_id == language.id && ResourceValue.supports_mode(resource, mode)
+    end
+
+    resource_values = Enum.filter(resources, matching_source?)
```

Comment:
Since we are not using matching_source? somewhere else, we can just place it inside Enum.find/2
```suggestion
    # Filter resources by language and mode.
    resource_values = Enum.filter(resources, fn resource ->
      resource.language_id == language.id && ResourceValue.supports_mode(resource, mode)
    end)
```

Reply:
Yes, nice suggestion! I applied here: https://github.com/turnhub/flow_runner/pull/247/commits/a9ca943c8a8295c40daead43dc9f2336a42f6570

Reply:
@joaothallis Using `Enum.find/2` looks way better, thanks 💯
However, this does not solve my original suggestion about the anonymous function 😅 Can we also include that?